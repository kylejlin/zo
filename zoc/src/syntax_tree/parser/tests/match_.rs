use super::*;

#[test]
fn match_() {
    let src = r#"
(
    match

    // Matchee
    0

    // Return arity
    3

    // Return val
    (for (500) 600)

    // Cases
    (
        (0 1)

        (3 400)

        (1 2)
    )
)"#;
    let tokens = lex(src).unwrap();
    let ost = parse(tokens).unwrap();
    insta::assert_debug_snapshot!(&ost);
}

#[test]
fn missing_return_arity() {
    let src = r#"
(
    match

    // Matchee
    0

    // Return val
    (for (500) 600)

    // Cases
    (
        (0 1)

        (3 400)

        (1 2)
    )
)"#;
    let tokens = lex(src).unwrap();
    let err = parse(tokens).unwrap_err();
    insta::assert_debug_snapshot!(&err);
}

#[test]
fn non_parenthesized_cases() {
    let src = r#"
(
    match

    // Matchee
    0

    // Return arity
    3

    // Return val
    (for (500) 600)

    // Unparenthesized case
    (0 1)
)"#;
    let tokens = lex(src).unwrap();
    let err = parse(tokens).unwrap_err();
    insta::assert_debug_snapshot!(&err);
}
