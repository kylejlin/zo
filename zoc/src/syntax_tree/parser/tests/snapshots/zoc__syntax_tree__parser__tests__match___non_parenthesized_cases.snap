---
source: zoc/src/syntax_tree/parser/tests/match_.rs
expression: "&err"
---
Some(
    Number(
        NumberLiteral {
            value: 0,
            span: (
                ByteIndex(
                    135,
                ),
                ByteIndex(
                    136,
                ),
            ),
        },
    ),
)
